tests/test_utils.py

```diff
@@ -172,3 +176,59 @@ def test_p_values_from_sample():
 
     ks_results = insertion_p_value(ns.insertion[nlive:-nlive], nlive, batch=1)
     assert ks_results['p-value'] > 0.05
+
+
+def test_credibility_interval():
```

Comment:
I'd suggest compactifying `test_credibility_interval` as follows. The parametrised tests ensure we don't miss any cases (currently we miss `return_covarianc==True and 'limit' in method`...).

I also changed the cov tests a bit, making them a bit more permissive such that they are more robust to different seeds and to a different number of iterations. After all, the values you tested cov against were not theory values, right? They just happened to work for that seed and for that number of iterations?

Something along the lines of this:

```python
@pytest.mark.parametrize('method', ['iso-pdf', 'equal-tailed',
                                    'lower-limit', 'upper-limit'])
@pytest.mark.parametrize('return_covariance', [True, False])
def test_credibility_interval(method, return_covariance):
    np.random.seed(0)
    normal_samples = np.random.normal(loc=2, scale=0.1, size=10000)
    ci = credibility_interval(
        normal_samples,
        level=0.84 if 'limit' in method else 0.68,
        method=method,
        return_covariance=return_covariance,
    )
    if return_covariance:
        ci, cov = ci
        assert np.ndim(cov) == 0 if 'limit' in method else 2
        assert np.all(np.abs(cov) < 1e-4)
        assert np.all(np.abs(cov) > 1e-8)
    if method in ['iso-pdf', 'equal-tailed']:
        lower, upper = ci
    else:
        upper = ci if 'upper' in method else 2.1
        lower = ci if 'lower' in method else 1.9
    assert_allclose(lower, 1.9, atol=0.02)
    assert_allclose(upper, 2.1, atol=0.02)
```

Reply:
This is great, I've replaced my function with this!

> After all, the values you tested cov against were not theory values, right? They just happened to work for that seed and for that number of iterations?

Yep